### ISSUE TITLE
Docs built

### ISSUE THREAD

**Issue by navidcy** [CliMA/ClimaOcean.jl#161]
At the moment the docs are build with GitHub actions and also via buildkite-ci for examples.

But we can't have two ways of building the docs. When GitHub actions build docs without the examples then it pushes it and it overwrites the other docs. However the buildkite-CI only runs when the PR has a particular label... 

The workflow is not robust and results in situations like the one described in #157...

**Comment by simone-silvestri** [comment]
I thought that #156 would solve this issue by nuking the GitHub actions build. Note that the docs are build always on main. The steps in buildkite read:
```
steps:
  - command: "buildkite-agent pipeline upload .buildkite/examples_build.yml "
    label: ":pipeline:"
    if: "build.pull_request.labels includes \"Run examples\" || build.branch == \"main\"" 
    agents:
      queue: new-central
```

**Comment by navidcy** [comment]
Yes, #156  should resolve this. Just wanted to document it for future.

**Comment by glwagner** [comment]
no way

**Comment by glwagner** [comment]
i don't see docs yet but maybe they are coming

**Comment by navidcy** [comment]
I’m worried there might be hosting/deploying issues. But they are built !!

**Comment by navidcy** [comment]
Still building... it takes 5-6hrs...

https://buildkite.com/clima/climaocean-examples/builds/755

**Comment by navidcy** [comment]
I’ll close it when I see the docs

**Comment by navidcy** [comment]
Hm... #156 docs+examples were OK but when merged I see this error:
https://buildkite.com/clima/climaocean-examples/builds/766#0192f3b5-7707-4829-b955-922fa0f1539d/188-313

I'm a bit in despair here. I don't know why all these issues pop up. Do we need to reconsider the buildkite pipeline? It'd be so good to have the examples in the docs and I'm very happy to put effort but it feels that errors appear stochastically? 

cc @simone-silvestri @Sbozzolo @glwagner

**Comment by glwagner** [comment]
> Hm... #156 docs+examples were OK but when merged I see this error: https://buildkite.com/clima/climaocean-examples/builds/766#0192f3b5-7707-4829-b955-922fa0f1539d/188-313
> 
> I'm a bit in despair here. I don't know why all these issues pop up. Do we need to reconsider the buildkite pipeline? It'd be so good to have the examples in the docs and I'm very happy to put effort but it feels that errors appear stochastically? Any insights are really welcomed!
> 
> cc @simone-silvestri @Sbozzolo @glwagner

I suspect the download was interrupted midway. I've seen this cause problems before. I think we want to use a `try/catch`, when there's an error loading the data, we should try to download it again. And we should encode a number of tries to use, something like 3 total tries before giving up. This needs another PR.

Sorry @navidcy part of the problem is that a lot of software remains to be written here. I think this is a sign that our system is brittle and buggy right now. The main thing we need to do is to make the software more robust and better.

**Comment by navidcy** [comment]
OK, good idea with the try-catch!

**Comment by glwagner** [comment]
If you look at the previous build you do indeed see an issue: https://buildkite.com/clima/climaocean-examples/builds/766#0192ef56-76dc-4e0a-9294-ba3598531c61/188-1072

You could test this with a try/catch in the example, ie if the build fails, catch the error, delete the existing file, and try again.

If that seems to work robustly we found the issue and we can write it in the source code.

**Comment by glwagner** [comment]
I can open a PR

**Comment by navidcy** [comment]
> I can open a PR

Yes please! ❤️

**Comment by simone-silvestri** [comment]
I think it might have been a problem of the server being down since it happened in all PRs together. Since we rely on automatic download we probably have to think how to deal with the fact that there might be interruptions to services we rely on

**Comment by glwagner** [comment]
For sure that is exactly what I am proposing to do; when loading a file fails, we assume it is because the file is corrupted due to partial download. Then we try to download again. We limit to a small number of retries, because sometimes server outage is persistent.

**Comment by glwagner** [comment]
Please let me know if that makes sense or if you have a better way to deal with this issue which is pretty common.

**Comment by Sbozzolo** [comment]
> I think it might have been a problem of the server being down since it happened in all PRs together. Since we rely on automatic download we probably have to think how to deal with the fact that there might be interruptions to services we rely on

What we do for this is the following: we split the test for downloading files and using files. In other words, we have a local copy of all the files needed to run a test on the cluster (so that we can always run), and the that downloads files could be a separate one (that can soft-file). For smaller files, we also rely on Julia's artifact system, which has a already redundancy built-in.

But I also agree on Greg's suggestion.

**Comment by navidcy** [comment]
OK, This is now resolved via #234.

Key factors towards the success:

Don't include 

```
withenv("GITHUB_REPOSITORY" => "github.com/CliMA/ClimaOceanDocumentation.git") do
     ....
end
```
which triggers Documenter to thing it should deploy via GitHub actions.


Also, we can ensure that Documenter deploys via Buildkite by providing the kwarg `deploy_config = Documenter.Buildkite()` to `deploydocs`.
